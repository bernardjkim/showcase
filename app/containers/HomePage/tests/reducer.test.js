// import { fromJS } from 'immutable';
import homePageReducer, { initialState } from '../reducer';
import {
  loadArticlesAll,
  loadArticlesAllError,
  loadArticlesAllSuccess,
} from '../actions';

describe('homePageReducer', () => {
  let state;
  beforeEach(() => {
    state = initialState;
  });

  it('returns the initial state', () => {
    expect(homePageReducer(state, {})).toMatchSnapshot();
  });

  it('handle the load articles all action', () => {
    expect(homePageReducer(state, loadArticlesAll())).toMatchSnapshot();
  });

  it('handle the load articles all success action', () => {
    const articles = [];
    expect(
      homePageReducer(state, loadArticlesAllSuccess(articles)),
    ).toMatchSnapshot();
  });

  it('handle the load articles all error action', () => {
    const error = new Error('Test Error');
    expect(
      homePageReducer(state, loadArticlesAllError(error)),
    ).toMatchSnapshot();
  });
});
