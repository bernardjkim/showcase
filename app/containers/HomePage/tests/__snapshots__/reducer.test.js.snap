// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePageReducer handles the createComment action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the createCommentError action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {},
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the createCommentSuccess action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [
      Immutable.Map {},
    ],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the likeArticle action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 2,
  },
}
`;

exports[`homePageReducer handles the likeArticleError action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {},
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the likeArticleSuccess action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the loadArticle action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "article": false,
}
`;

exports[`homePageReducer handles the loadArticleError action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {},
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the loadArticleSuccess action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": Immutable.Map {},
}
`;

exports[`homePageReducer handles the loadComments action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the loadCommentsError action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {},
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer handles the loadCommentsSuccess action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": Immutable.Map {
    "title": "google",
    "uri": "https://google.com",
    "github": "",
    "description": "Search Engine",
    "image": "1548552844157-Screen Shot 2019-01-26 at 5.32.10 PM.png",
    "comments": Immutable.List [],
    "likes": 1,
  },
}
`;

exports[`homePageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": false,
}
`;
